### DISCUSSION TITLE
Remove collector from the picture

### DISCUSSION THREAD

**Discussion by Shinjiru-stack** [glowroot/glowroot#910] [Q&A] [answered]
Can we remove collector from the application and directly save incoming trace data to database like MongoDB?

**Comment by Never-lab** [comment, marked as answer]
Not supported today. Agents either write locally (**embedded**, H2) or send gRPC to **Glowroot Central** (Cassandra). There is no path to point the agent at MongoDB (or another DB) and skip the collector.

Closest options: embedded, or Central + Cassandra. Feature requests → [Discussions → Ideas](https://github.com/glowroot/glowroot/discussions/categories/ideas). Background: [What Glowroot does not do](https://github.com/glowroot/glowroot/wiki/What-Glowroot-does-not-do).
